INSTALL.md

```diff
@@ -0,0 +1,87 @@
+# Idris Installation Guide
+
+Copyright (C) 2015, The Idris Community
+
+This file provides information over obtaining and installing Idris.
+
+## Installing Idris from Hackage
+
+This repository represents the latest development version of the language,
+and may contain bugs that are being actively worked on.
+For those who wish to use a more stable version of Idris please consider
+installing the latest version that has been released on Hackage.
+Installation instructions for various platforms can be [found on the Idris Wiki](https://github.com/idris-lang/Idris-dev/wiki/Installation-Instructions).
+
+## Installing Development Versions
+
+If you like to work against the latest development version, please consider
+using Cabal Sandboxes to minimise disruption to your local Haskell setup.
+Instructions for installing Idris HEAD within a cabal sandbox are
+[available on the Idris Wiki](https://github.com/idris-lang/Idris-dev/wiki/Installing-an-Idris-Development-version-in-a-sandbox).
+
+To configure, edit config.mk. The default values should work for most people.
+
+Idris is built using a Makefile common targets include:
+
+* `make` This will install everything using cabal and
+typecheck the libraries.
+* `make test` This target execute the test suite.
+* `make relib` This target will typecheck and recompile the standard library.
+
+Idris has an optional buildtime dependency on the C library
+`libffi`. If you would like to use the features that it enables, make
+sure that it is compiled for the same architecture as your Haskell
+compiler (e.g. 64 bit libraries for 64 bit ghc). By default, Idris
+builds without it. To build with it, pass the flag `-f FFI`.
+
+A secondary optional buildtime dependency is on `libGMP`, this allows
+better support for numeric operations. As with `libFFI` Idris builds
+with out it, to enable `GMP` support use the flag `-f GMP`
```

Comment:
It's not terribly important, but you can build GHC with GMP or with a
native Haskell integer library. I think the reason was licensing concerns?
So you could conceivably not have GMP when you compiled Idris.

On Thu, Dec 10, 2015 at 10:34 AM Leif Warner <notifications@github.com>
wrote:

> In INSTALL.md
> <https://github.com/idris-lang/Idris-dev/pull/2852#discussion_r47264775>:
>
> > +Idris is built using a Makefile common targets include:
> > +
> > +* `make` This will install everything using cabal and
> > +typecheck the libraries.
> > +* `make test` This target execute the test suite.
> > +* `make relib` This target will typecheck and recompile the standard library.
> > +
> > +Idris has an optional buildtime dependency on the C library
> > +`libffi`. If you would like to use the features that it enables, make
> > +sure that it is compiled for the same architecture as your Haskell
> > +compiler (e.g. 64 bit libraries for 64 bit ghc). By default, Idris
> > +builds without it. To build with it, pass the flag `-f FFI`.
> > +
> > +A secondary optional buildtime dependency is on `libGMP`, this allows
> > +better support for numeric operations. As with `libFFI` Idris builds
> > +with out it, to enable `GMP` support use the flag `-f GMP`
>
> Yeah, these are optional deps of the produced idris compiler. I think GMP
> is always a "buildtime" dep, as GHC itself depends on it (the Idris binary
> links to it on my system; compiled Idris programs link to it since I built
> idris with the -fGMP flag)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idris-lang/Idris-dev/pull/2852/files#r47264775>.
>